### PR TITLE
FM-36-Backend-pagination

### PR DESCRIPTION
**Pull request by gonzur75**
# Description
Backend pagination setup. [Youtrack](https://azeno.youtrack.cloud/issue/FM-36/Backend-pagination) 
Build in  [LimitOffsetPagination](https://www.django-rest-framework.org/api-guide/pagination/#limitoffsetpagination) class was chosen for pagination with only max_limit setup to 100

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The related memme has been added
pagination

### CODE REVIEW

File: flashcards/backend/decks/tests/test_pagination.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from decks import factories
+from decks.pagination import DecksPagination
+from decks.viewsets import DeckViewSet
+
+
+def test_deck_list_pagination(api_rf, db):
```
Comment (bartoszbaran2) [inline flashcards/backend/decks/tests/test_pagination.py:6]: correct me if I'm wrong but you use api_rf fixture which comes from quiz app, can we use fixtures from another apps?

Comment (martynawitkowska) [reply flashcards/backend/decks/tests/test_pagination.py:6]: Nope

Comment (gonzur75) [reply flashcards/backend/decks/tests/test_pagination.py]: I have changed it to correct fixture now. Thanks

**Review by habibuga — APPROVED** [review]
Created faked decks and tested with Postman. Pagination works as expected and sorts decks by popularity. Good job!
As I understood the option with getting the page size from the Front End will be developed later, correct?

File: flashcards/backend/decks/viewsets.py [hunk 2]
```diff
@@ -11,8 +11,8 @@ class DeckViewSet(viewsets.ModelViewSet):
     serializer_class = DeckSerializer
 
     def list(self, request, *args, **kwargs):
-        decks = self.serializer_class(self.get_queryset().order_by("-rating"), many=True)
```
[the remark this answers is not in the document]
Comment (gonzur75) [reply flashcards/backend/decks/viewsets.py:14]: You are commenting an  code from master branch, which i have modified. In my code paginate_queryset() is called right after sorting. You can see that in /rest_framework/mixins.py file, class ListModelMixin. I'm getting that by calling super method `return super().list(request, *args, **kwargs)`. But of course i might be doing something wrong here let me know your thoughts on that

**Review by gonzur75 — COMMENTED** [review]
Witalij you get custom amount of records by adjusting params ('limit' for amount of records and 'offset' for initial record) up to our max_limit 100 eg. `/v1/decks/?limit=80&offset=20`

**Review by aleksanderkluska — APPROVED** [review]
Looks good for me, great job.

**Review by bartoszbaran2 — APPROVED** [review]
approved

**Pull request merged**